src/chrdev.rs

```diff
@@ -142,8 +142,13 @@ unsafe extern "C" fn read_callback<T: FileOperations>(
         Err(e) => return e.to_kernel_errno() as c_types::c_ssize_t,
     };
     let f = &*((*file).private_data as *const T);
-    // TODO: Pass offset to read()?
-    match f.read(&mut data) {
+    // No FMODE_UNSIGNED_OFFSET support, so offset must be in [0, 2^63).
+    // See discussion in #113
+    let positive_offset = match (*offset).try_into() {
+        Ok(v) => v,
+        Err(_) => return Error::EINVAL.to_kernel_errno() as c_types:c_ssize_t,
```

Comment:
Can't use `into` here because we're defining `c_ssize_t` as `isize`, and Rust worries that pointer size might be as small as 16 bits: https://doc.rust-lang.org/src/core/num/mod.rs.html#5186-5191

Happy to deal with this later in some more elegant way.